src/lib/pipeline/wmo/group/index.js

```diff
@@ -27,22 +32,58 @@ class WMOGroup extends THREE.Mesh {
       uvs[index * 2 + 1] = textureCoords[index][1];
     });
 
+    data.MONR.normals.forEach(function(normal, index) {
+      normals[index * 3] = normal[0];
+      normals[index * 3 + 1] = normal[2];
+      normals[index * 3 + 2] = -normal[1];
+    });
+
+    if ('MOCV' in data) {
+      data.MOCV.colors.forEach(function(color, index) {
+        colors[index * 3] = color.r / 255.0;
+        colors[index * 3 + 1] = color.g / 255.0;
+        colors[index * 3 + 2] = color.b / 255.0;
+        alphas[index] = color.a / 255.0;
+      });
+    } else if (this.indoor) {
+      // Default indoor vertex color: rgba(0.5, 0.5, 0.5, 1.0)
+      data.MOVT.vertices.forEach(function(vertex, index) {
+        colors[index * 3] = 127.0/255.0;
+        colors[index * 3 + 1] = 127.0/255.0;
+        colors[index * 3 + 2] = 127.0/255.0;
+        alphas[index] = 1.0;
+      });
+    }
+
     const indices = new Uint32Array(data.MOVI.triangles);
 
     const geometry = this.geometry = new THREE.BufferGeometry();
     geometry.setIndex(new THREE.BufferAttribute(indices, 1));
     geometry.addAttribute('position', new THREE.BufferAttribute(positions, 3));
+    geometry.addAttribute('normal', new THREE.BufferAttribute(normals, 3));
     geometry.addAttribute('uv', new THREE.BufferAttribute(uvs, 2));
 
+    // TODO: Perhaps it is possible to directly use a vec4 here? Currently, color + alpha is
+    // combined into a vec4 in the material's vertex shader. For some reason, attempting to
+    // directly use a BufferAttribute with a length of 4 resulted in incorrect ordering for the
+    // values in the shader.
+    geometry.addAttribute('color', new THREE.BufferAttribute(colors, 3));
+    geometry.addAttribute('alpha', new THREE.BufferAttribute(alphas, 1));
```

Comment:
Yeah, sounds good. I couldn't tell if the issue was THREE trying to munge attributes named 'color' in a weird way, or if perhaps the values were perhaps going like this: [0, 1, 2, 3] => [3, 0, 1, 2]